unyt/array.py

```diff
@@ -843,21 +843,14 @@ def in_units(self, units, equivalence=None, **kwargs):
                         RuntimeWarning,
                         stacklevel=2,
                     )
-            new_dtype = np.dtype("f" + str(dsize))
-            conversion_factor = new_dtype.type(conversion_factor)
-            ret = np.asarray(self.ndview * conversion_factor, dtype=new_dtype)
+            new_dtypekind = "c" if self.dtype.kind == "c" else "f"
+            new_dtype = np.dtype(new_dtypekind + str(dsize))
+            ret = np.asarray(self.v * conversion_factor, dtype=new_dtype)
             if offset:
-                np.subtract(ret, offset, ret)
```

Comment:
It's not clear to me why these lines needed to be changed here.

Reply:
I seem to remember having trouble with the `bypass_validation=True` during the prototyping process. Reverting these lines appears to be acceptable.